applications/callflow/src/module/cf_resources.erl

```diff
@@ -341,3 +344,26 @@ handle_channel_destroy(<<"loopback", _/binary>>, _JObj) ->
     {<<"TRANSFER">>, 'ok'};
 handle_channel_destroy(_, JObj) ->
     {kz_call_event:hangup_cause(JObj), kz_call_event:hangup_code(JObj)}.
+
+-spec get_privacy_prefs(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs(Call) ->
+    lager:debug("Checking inception of call"),
+    case kapps_call:inception(Call) of
+        'undefined' -> get_privacy_prefs_from_endpoint(Call);
+        _Else -> []
+    end.
+
+-spec get_privacy_prefs_from_endpoint(kapps_call:call()) -> kz_term:proplist().
+get_privacy_prefs_from_endpoint(Call) ->
+    lager:debug("Call is outbound, checking caller_id_outbound_privacy value"),
+    {'ok', Endpoint} = kz_endpoint:get(Call),
+    case kz_json:get_value([<<"caller_id_options">>, <<"outbound_privacy">>], Endpoint) of
+        'undefined' ->
+            [];
+        %% can't call kapps_call_command:privacy/2 with Mode = <<"none">>
+        <<"none">>=NoneMode ->
+            cf_util:ccvs_by_privacy_mode(NoneMode);
+        Mode ->
+            kapps_call_command:privacy(Mode, Call),
```

Comment:
why is this being called here ?

Reply:
I assume you mean the call to kapps_call_command:privacy. This is mimicking the behaviour from the cf_privacy module and the caller ID attributes won't be hidden unless it's called.

Reply:
i don't think this should be called here, we are creating properties for the amqp payload. we may need to handle this on stepswitch and/or ecallmgr_fs_bridge

Reply:
Okay, valid point. Seeing as it's already merged, how would you like to proceed? Should I change it and create a new pull request?